http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Http2IT.java

```diff
@@ -0,0 +1,219 @@
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.junit5.Checkpoint;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+@QuarkusScenario
+@ExtendWith(VertxExtension.class)
+public class Http2IT {
+    private static final Logger LOG = Logger.getLogger(Http2IT.class);
+    @QuarkusApplication(classes = { MorningResource.class }, properties = "http2.properties")
+    static RestService app = new RestService();
+
+    private static URILike baseUri;
+    static private Vertx vertx;
+
+    private static final String BASE_ENDPOINT = "/morning";
+    private static final String GREETING = "Buenos dias";
+
+    private static HttpClient httpClient;
+
+    @BeforeAll
+    static void setUp() {
+        baseUri = app.getURI();
+        vertx = Vertx.vertx();
+    }
+
+    @Test
+    @DisplayName("HttpClient Vertx compatibility HTTP/1.1 Test")
+    public void httpClientVertxHttp1(VertxTestContext vertxTestContext) {
+        Checkpoint requestCheckpoint = vertxTestContext.checkpoint(2);
+        assertNotNull(vertx);
+        httpClient = vertx.createHttpClient();
+        assertNotNull(httpClient);
+        String expectedHTTP1Version = "HTTP_1_1";
+
+        vertxTestContext.verify(() -> httpClient.request(HttpMethod.GET, baseUri.getPort(), baseUri.getHost(), BASE_ENDPOINT)
+                .compose(req -> req.send()
+                        .compose(httpClientResponse -> {
+                            assertEquals(HttpStatus.SC_OK, httpClientResponse.statusCode());
+                            assertEquals(expectedHTTP1Version, httpClientResponse.version().toString());
+                            requestCheckpoint.flag();
+                            return httpClientResponse.body();
+                        }))
+                .onSuccess(body -> {
+                    LOG.info("Got data " + body.toString("ISO-8859-1"));
+                    assertThat("Body response", body.toString().contains(GREETING));
+
+                    requestCheckpoint.flag();
+
+                }).onFailure(Throwable::printStackTrace));
+
+    }
+
+    @Test
+    @DisplayName("HttpClient Vertx HTTP/2 protocol Test with ssl, alpn, trustAll true and verifyHost false")
+    public void httpClientVertxHttp2(VertxTestContext vertxTestContext) {
+
+        Checkpoint requestCheckpoint = vertxTestContext.checkpoint(2);
+        assertNotNull(vertx);
+        httpClient = vertx.createHttpClient(new HttpClientOptions()
+                .setSsl(true)
+                .setUseAlpn(true)
+                .setTrustAll(true)
+                .setVerifyHost(false)
+                .setProtocolVersion(HttpVersion.HTTP_2));
+        assertNotNull(httpClient);
+        vertxTestContext.verify(() -> httpClient.request(HttpMethod.GET, 8443, baseUri.getHost(), BASE_ENDPOINT)
```

Comment:
`8443`? I think you should use `@QuarkusApplication(ssl = true)` and get port from `app.getURI(HTTPS)`